django/db/models/functions/comparison.py

```diff
@@ -74,6 +74,23 @@ def as_oracle(self, compiler, connection, **extra_context):
         return self.as_sql(compiler, connection, **extra_context)
 
 
+class Collate(Func):
+    function = 'COLLATE'
+    template = '%(expressions)s %(function)s %(collation)s'
+
+    def __init__(self, *expressions, **extra):
+        if len(expressions) != 2:
+            raise ValueError('Collate must take exactly two arguments')
+        collation = expressions[-1]
+        expressions = [expressions[0]]
+        super().__init__(*expressions, **extra)
```

Comment:
I _think_ you can achieve the same thing while making the signature self documenting with

```suggestion

    def __init__(self, expression, collation, **extra):
        self.collation = collation
        super().__init__(expression, **extra)
```